### ISSUE TITLE
[upgrade test failure] Restart [Disruptive] should restart all nodes and ensure all nodes and pods recover

### ISSUE THREAD

**Issue by ericchiang** [kubernetes/kubernetes#50797]
Opening this since this seems slightly different from https://github.com/kubernetes/kubernetes/issues/46651

/cc @kubernetes/sig-node-bugs 

This tests has been consistently failing on a lot of the upgrade tests:

https://k8s-testgrid.appspot.com/master-upgrade#gke-cvm-1.7-gci-master-upgrade-master
https://k8s-testgrid.appspot.com/master-upgrade#gke-cvm-1.7-gci-master-upgrade-cluster
https://k8s-testgrid.appspot.com/master-upgrade#gke-gci-1.7-cvm-master-upgrade-master
https://k8s-testgrid.appspot.com/master-upgrade#gke-gci-1.7-cvm-master-upgrade-cluster
https://k8s-testgrid.appspot.com/master-upgrade#gke-gci-1.7-gci-master-upgrade-master
https://k8s-testgrid.appspot.com/master-upgrade#gke-gci-1.7-gci-master-upgrade-cluster

What's really weird is the error message most of them spit out

```
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/restart.go:92
Expected error:
    <*errors.errorString | 0xc422322770>: {
        s: "couldn't find -1 nodes within 20s; last error: expected to find -1 nodes but found only 3 (20.007524987s elapsed)",
    }
    couldn't find -1 nodes within 20s; last error: expected to find -1 nodes but found only 3 (20.007524987s elapsed)
not to have occurred
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/restart.go:77
```

https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gke-cvm-new-cvm-master-upgrade-cluster/128#k8sio-restart-disruptive-should-restart-all-nodes-and-ensure-all-nodes-and-pods-recover

Really hard to be clear who owns this tests, going to tag sig-node until there's further evidence otherwise.

cc @kubernetes/kubernetes-release-managers @mbohlool

**Comment by wojtek-t** [comment]
https://github.com/kubernetes/test-infra/pull/4086#issuecomment-322895571 should fix this

**Comment by wojtek-t** [comment]
@krzyzacy - FYI

**Comment by krzyzacy** [comment]
The tests has wrong `--num-nodes` are passing now, however seems the actual upgrade test is still failing.

/unassign
/assign @ericchiang 
feel free to reassign to interested parties.

**Comment by ericchiang** [comment]
This upgrade test is much healthier since the fix went in and those failures look related to overall test environment issues. Closing. Will open another issue if I identify anything specific about this test.

Thanks @krzyzacy !

**Comment by ericchiang** [comment]
Reopening since this is still failing on on our or upgrade tests.

https://k8s-testgrid.appspot.com/master-upgrade#gke-gci-1.7-gci-master-upgrade-cluster-new

We're blocked on that test not producing logs though (https://github.com/kubernetes/kubernetes/issues/52578).

**Comment by yujuhong** [comment]
```
couldn't find -1 nodes within 20s; last error: expected to find -1 nodes but found only 3 (20.008632278s elapsed)",
```
The expected number of nodes "-1" was the value of `framework.TestContext.CloudConfig.NumNodes`
This recent PR changed how this values is set in GKE: https://github.com/kubernetes/test-infra/pull/4500
There are other tests with similar failure signature too. 
/cc @krzyzacy @zmerlynn

**Comment by krzyzacy** [comment]
It was like that before my PR, @zmerlynn seems `--num-nodes=3` is passed properly but it's still insufficient to fix the issue? I'll take another look.

**Comment by krzyzacy** [comment]
Before #4500 the entire upgrade was failing (if you scroll to the right of the testgrid page, that fixed `UpgradeTest` row).

So this test was passing in both https://k8s-testgrid.appspot.com/master-upgrade#gke-gci-1.7-gci-master-upgrade-cluster and https://k8s-testgrid.appspot.com/master-upgrade#gke-gci-1.7-gci-master-upgrade-master, @crimsonfaith91 can you double check that if there's any difference (need to be cherrypicked back to 1.7)  between 1.7 and 1.8?

**Comment by krzyzacy** [comment]
fixed by https://github.com/kubernetes/test-infra/pull/4617
/close

**Comment by crimsonfaith91** [comment]
Thanks, @krzyzacy!
